src/main/java/org/killbill/billing/plugin/adyen/client/model/PaymentModificationResponse.java

```diff
@@ -31,11 +36,17 @@ public PaymentModificationResponse(final String response, final String pspRefere
         this.response = response;
     }
 
+    public PaymentModificationResponse(final String pspReference, final AdyenCallErrorStatus adyenCallErrorStatus) {
+        this.pspReference = pspReference;
+        this.adyenCallErrorStatus = Optional.of(adyenCallErrorStatus);
+    }
+
     public PaymentModificationResponse(final String response, final String pspReference) {
         this(response, pspReference, new HashMap<Object, Object>());
     }
 
-    public PaymentModificationResponse() {
+    public boolean isSuccess() {
```

Comment:
good catch! That's right. The main problem in the plugin is the amount of unneeded code and layers => we have to implement the same method multiple times.